Assignment4/patricifrian-190112/task07-updated.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1zZUeTCp-VpsiVugt0e6UXlnV26GOvBAb
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+rdfs = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+# Using RDFLib
+print("Print results using RDFLib")
+for s, p, o in g.triples((None, rdfs.subClassOf, ns.LivingThing)):
+    print(s)
+    for q, p, o in g.triples((None, rdfs.subClassOf, s)):
+      print("Subclass:"+q)
+
+# Using SPARQL
+q1 = prepareQuery('''
+  SELECT ?Subject ?Subclass WHERE {
+    ?Subject rdfs:subClassOf ns:LivingThing .
+    OPTIONAL {?Subclass rdfs:subClassOf ?Subject .}
```

Comment:
what about the subclasses of subclasses of subclasses? You need to design a generic query.